kgforge/specializations/models/rdf/service.py

```diff
@@ -272,3 +338,109 @@ def traverse_properties(properties) -> Tuple[Dict, Dict]:
         context.update({key: terms[key] for key in sorted(terms)})
 
         return {"@context": context} if len(context) > 0 else None
+
+    def transitive_load_shape_graph(
+        self, graph_id: str, schema_id: str, node_shape_uriref: URIRef = None
+    ):
+        schema_graph = self.load_shape_graph(graph_id, schema_id)
+        self._imported.append(schema_id)
```

Comment:
I feel like this should only be done when the whole operation is successful, otherwise if an exception is thrown mid-way, the _imported content is not right

Reply:
I moved it to the end of this method